src/components/BeverageForm.js

```diff
@@ -29,37 +30,37 @@ type Props = {|
   submitButtonLabel: string,
 |};
 
-type InjectedProps = FormProps;
+type InjectedProps = {
+  ...FormProps,
+  availabilityStore: DAOEntityStore<Availability>,
+  glassStore: DAOEntityStore<Glass>,
+  srmStore: DAOEntityStore<Srm>,
+  styleStore: DAOEntityStore<Style>,
+};
 
 @form()
+@withDAOEntityStore('availabilityStore', DAOApi.AvailabilityDAO)
+@withDAOEntityStore('srmStore', DAOApi.SrmDAO, {
+  orderBy: [{ column: 'hex', direction: 'desc' }],
+})
+@withDAOEntityStore('glassStore', DAOApi.GlassDAO)
+@withDAOEntityStore('styleStore', DAOApi.StyleDAO)
```

Comment:
I still do not like this pattern. 

1. There really isn't a good reason why we want to inject these stores. What benefit do we get from the added complexity?
2. Let's say I have an ID I need to pass into one of the stores for query filter? It makes the third parameter pretty useless since we now have to call a setter in componentWillMount

We need to figure out where the "business logic" for components should live. Do we want to use containers or use stores as selectors?

Reply:
1. it reduces `componentDidMount() { store.dispose() } in every component where we use stores with dao.subscribe()
2. we can do `(props) => props.id`  in the selector, why component will mount?
3. yeah..let's discuss that in slack....In general, I don't like containers(in form as they are in redux) in mobx, since we still have to decorate our views with @observer and operate with obsrvable dat a there, this makes our views kind of coupled to mobx anyways.... so if you want to move all bussiness logic from views I prefer to use stores as the source of truth. The only thing that I complain about I just don't want to see componentWillUnmount() { ...} in every component.

Reply:
1. I still agree that we need that. I just don't think we need to limit the store to being an entity store.
2.  I don't really like the pattern of doing selectors in an attribute. That could get pretty messy with a complex screen
3. Yeah, I'd like the logic put into stores. That way if we ever wanted to write unit tests it would be much cleaner than using components.